### ISSUE TITLE
Compatibility issues because of using String+Bytes instead of SBytes

### ISSUE THREAD

**Issue by sibbl** [RNCryptor/RNCryptor-cs#7]
I encountered the problem that I had a string, which was a encrypted and then gzipped JSON string. This was done on another platform and can't be decrypted correctly using the current code of this repo. The problem I found is that Java bytes are signed, whereas CSharp bytes are unsigned. Maybe one should consider changing this in this project in general.

I am no master of bytes and sbytes and rarely use them, but could solve my special problem by using the following method for `decryptAesCbcPkcs7` in the decryptor and return `sbytes[]` instead of a `string`.

```
private sbyte[] decryptAesCbcPkcs7 (byte[] encrypted, byte[] key, byte[] iv)
{
  var aes = Aes.Create();
  aes.Mode = CipherMode.CBC;
        aes.Padding = PaddingMode.PKCS7;
  var decryptor = aes.CreateDecryptor(key, iv);

        var result = new List<sbyte>();
        using (var msDecrypt = new MemoryStream(encrypted))
  {
    using (var csDecrypt = new CryptoStream(msDecrypt, decryptor, CryptoStreamMode.Read))
            {
                using (var ms = new MemoryStream())
                {
                    csDecrypt.CopyTo(ms);
                    result.AddRange(ms.ToArray().Select(r => (sbyte) r));
                }
            }
  }

    return result.ToArray();
}
```

This makes sure that the MemoryStream doesn't generate a `byte[]` but a `sbyte[]` and doesn't return a string after decrypting, but also a `sbyte[]`. I don't know why exactly but I could convert this `sbyte[]` to a `byte[]` afterwards and pass it to the un-gzip function I am using (which is not part of this project, but should help understanding why I get non-simple-plain-chars). 

I could provide you a unit test if it may help but I don't know how you consider refactoring the code to use sbytes and bytes. I don't want to make a pull request of a completely restructured project ;)

**Comment by evildivine** [comment]
sbyte solution worked for me in decrypting part .... now I need to encrypt it back and there is probably similar issue :( 

i am trying to achieve iOS <-> Windows implementation

**Comment by rnapier** [comment]
I believe this is resolved with #8
